### PR TITLE
add full PPR e2e tests

### PR DESCRIPTION
**Pull request by ztanner**
This copies over the E2E tests that were added to `vercel/vercel` in https://github.com/vercel/vercel/pull/10808 and https://github.com/vercel/vercel/pull/10823, with some minor adjustments to reflect the differences in cache behavior between start & deploy.

### CODE REVIEW

**Comment by ztanner** [comment]
Current dependencies on/for this PR:
* canary
  * **PR #59025** <a href="https://app.graphite.dev/github/pr/vercel/next.js/59025?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/vercel/next.js/59025?utm_source=stack-comment).

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 8875a14cf78b55e4e0d8713c6d41328292b70420

`pnpm test test/integration/preload-viewport/test/index.test.js` 
- Prefetching Links in viewport > production mode > should prefetch with non-bot UA

<details>
<summary>Expand output</summary>

  ● Prefetching Links in viewport › production mode › should prefetch with non-bot UA

    expect(received).toHaveLength(expected)

    Expected length: 1
    Received length: 0
    Received array:  []

      129 |         )
      130 |         const links = await browser.elementsByCss('link[rel=prefetch]')
    > 131 |         expect(links).toHaveLength(1)
          |                       ^
      132 |       } finally {
      133 |         if (browser) await browser.close()
      134 |       }

      at Object.toHaveLength (integration/preload-viewport/test/index.test.js:131:23)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 11-28-add_full_PPR_e2e_tests | Change |
| - | - | - | - |
| buildDuration | 10.4s | 10.3s | N/A |
| buildDurationCached | 5.8s | 6.5s | ⚠️ +641ms |
| nodeModulesSize | 199 MB | 199 MB | ✓ |
| nextStartRea..uration (ms) | 422ms | 417ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 11-28-add_full_PPR_e2e_tests | Change |
| - | - | - | - |
| 199-HASH.js gzip | 28.7 kB | 28.7 kB | N/A |
| 3f784ff6-HASH.js gzip | 53.3 kB | 53.3 kB | ✓ |
| 494.HASH.js gzip | 180 B | 181 B | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 241 B | 238 B | N/A |
| main-HASH.js gzip | 31.7 kB | 31.7 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | ✓ |
| Overall change | 100 kB | 100 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 11-28-add_full_PPR_e2e_tests | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 11-28-add_full_PPR_e2e_tests | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 182 B | 181 B | N/A |
| amp-HASH.js gzip | 501 B | 503 B | N/A |
| css-HASH.js gzip | 322 B | 323 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | ✓ |
| edge-ssr-HASH.js gzip | 253 B | 255 B | N/A |
| head-HASH.js gzip | 348 B | 347 B | N/A |
| hooks-HASH.js gzip | 369 B | 368 B | N/A |
| image-HASH.js gzip | 4.27 kB | 4.27 kB | N/A |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.61 kB | 2.6 kB | N/A |
| routerDirect..HASH.js gzip | 311 B | 311 B | ✓ |
| script-HASH.js gzip | 384 B | 383 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.17 kB | 3.17 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 11-28-add_full_PPR_e2e_tests | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 483 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 11-28-add_full_PPR_e2e_tests | Change |
| - | - | - | - |
| index.html gzip | 529 B | 529 B | ✓ |
| link.html gzip | 540 B | 542 B | N/A |
| withRouter.html gzip | 525 B | 524 B | N/A |
| Overall change | 529 B | 529 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 11-28-add_full_PPR_e2e_tests | Change |
| - | - | - | - |
| edge-ssr.js gzip | 92.5 kB | 92.5 kB | N/A |
| page.js gzip | 145 kB | 145 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 11-28-add_full_PPR_e2e_tests | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 624 B | 626 B | N/A |
| middleware-r..fest.js gzip | 150 B | 151 B | N/A |
| middleware.js gzip | 35.6 kB | 35.6 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 1.92 kB | 1.92 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 11-28-add_full_PPR_e2e_tests | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 167 kB | 167 kB | ✓ |
| app-page-exp..prod.js gzip | 93.5 kB | 93.5 kB | ✓ |
| app-page-tur..prod.js gzip | 94.3 kB | 94.3 kB | ✓ |
| app-page-tur..prod.js gzip | 88.8 kB | 88.8 kB | ✓ |
| app-page.run...dev.js gzip | 137 kB | 137 kB | ✓ |
| app-page.run..prod.js gzip | 88.2 kB | 88.2 kB | ✓ |
| app-route-ex...dev.js gzip | 23.8 kB | 23.8 kB | ✓ |
| app-route-ex..prod.js gzip | 16.5 kB | 16.5 kB | ✓ |
| app-route-tu..prod.js gzip | 16.5 kB | 16.5 kB | ✓ |
| app-route-tu..prod.js gzip | 16 kB | 16 kB | ✓ |
| app-route.ru...dev.js gzip | 23.2 kB | 23.2 kB | ✓ |
| app-route.ru..prod.js gzip | 16 kB | 16 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.64 kB | 9.64 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-turbo...prod.js gzip | 21.9 kB | 21.9 kB | ✓ |
| pages.runtim...dev.js gzip | 22.5 kB | 22.5 kB | ✓ |
| pages.runtim..prod.js gzip | 21.9 kB | 21.9 kB | ✓ |
| server.runti..prod.js gzip | 49.2 kB | 49.2 kB | ✓ |
| Overall change | 925 kB | 925 kB | ✓ |

</details>

</details>
Commit: 8875a14cf78b55e4e0d8713c6d41328292b70420

File: test/e2e/app-dir/ppr-full/app/dynamic/force-dynamic/page.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React, { Suspense } from 'react'
+import { Dynamic } from '../../../components/dynamic'
+
+export const dynamic = 'force-dynamic'
+export const revalidate = 60
```
Comment (ijjk) [inline test/e2e/app-dir/ppr-full/app/dynamic/force-dynamic/page.jsx:5]: What is this checking for? Doesn't the `force-dynamic` cancel out our `revalidate: 60`?

**Review by ijjk — DISMISSED** [review]

**Review by ijjk — APPROVED** [review]

**Review by acdlite — APPROVED** [review]

**Pull request merged**